lib/core/util.c

```diff
@@ -945,3 +946,60 @@ void h2o_cleanup_thread(void)
     h2o_mem_clear_recycle(&h2o_http2_wbuf_buffer_prototype.allocator);
     h2o_mem_clear_recycle(&h2o_socket_buffer_prototype.allocator);
 }
+
+void h2o_generate_uuidv4(char *buf)
+{
+    // RFC-4122 4.4. Algorithms for Creating a UUID from Truly Random or Pseudo-Random Numbers
+
+    // 4.2.1 Layout And Byte Order
+    struct rfc4122 {
+        uint32_t time_low;
+        uint16_t time_mid;
+        uint16_t time_hi_and_version;
+        uint8_t clock_seq_hi_and_reserved;
+        uint8_t clock_seq_low;
+        uint8_t node[6];
+    } uuid;
+    _Static_assert(sizeof(uuid) == 16, "size of UUID structure must be 16");
+
+    if (RAND_bytes((void*)&uuid, sizeof(uuid)) != 1) {
+        h2o_fatal("RAND_bytes failed");
+    }
```

Comment:
This is more of an editorial choice, but please use `ptls_openssl_random_bytes` instead, as we do elsewhere.